File: 2023-2024/sourcecode/mod3/lab9/RepositoryBasicsJava/app/src/main/java/com/example/repositorybasicsjava/viewmodel/UserViewModel.java
package com.example.repositorybasicsjava.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.example.repositorybasicsjava.model.User;
import com.example.repositorybasicsjava.repository.UserRepository;

import java.util.List;

public class UserViewModel extends ViewModel {

    private UserRepository userRepository;
    private final MutableLiveData<List<User>> usersList = new MutableLiveData<>();
    public LiveData<List<User>> getUsersList() {
        return usersList;
    }

    public UserViewModel() {
        userRepository = new UserRepository();
        loadUsers();
    }

    private void loadUsers(){
        usersList.setValue(userRepository.getUsers());
    }

}

File: 2023-2024/sourcecode/mod3/lab9/RepositoryBasicsCompose/app/src/main/java/com/example/repositorybasicscompose/viewmodel/UserViewModel.kt
package com.example.repositorybasicscompose.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.example.repositorybasicscompose.model.User
import com.example.repositorybasicscompose.repository.UserRepository
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.launch

class UserViewModel() : ViewModel() {

    private val userRepository = UserRepository()

    private val _usersList = MutableStateFlow<List<User>>(emptyList())
    val usersList: StateFlow<List<User>> get() = _usersList

    init {
        loadUsers()
    }

    private fun loadUsers() {
        viewModelScope.launch {
            _usersList.value = userRepository.getUsers()
        }
    }
}